### PR TITLE
server ms1

### PR DESCRIPTION
**Pull request by CecileJiaqiWang**

### CODE REVIEW

File: server/middleware/validation.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+exports.isValidCoordinate = (req, res, next) => {
+    if (req.body.hasOwnProperty("coord")) {
+        let user_coord = req.body.coord;
+        if (user_coord["lat"] <= 90 && user_coord["lat"] >= -90 && user_coord["lng"] <= 180 && user_coord["lng"] >= -180) {
```
Comment (yorhodes) [inline server/middleware/validation.js]: nice!

File: server/models/storyModel.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    result(err, null);
+                } else {
+                    console.log(record.insertId);
+                    result(null, record.insertId);
+                }
+            });
+    }
+
+    static getStory(result) {
+        pool.query("SELECT * FROM stories ORDER BY ts DESC LIMIT 10",
```
Comment (yorhodes) [inline server/models/storyModel.js]: cool

File: server/controllers/storyController.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    if (!newStory.storyID) {
+        res.status(400).send({ error:true, message: 'Please provide story ID.' });
+    } else if (!newStory.gridID) {
+        res.status(400).send({ error:true, message: 'Please provide grid ID.' });
+    } else if (!newStory.ts) {
+        res.status(400).send({ error:true, message: 'Please provide timestamp.' });
+    } else if (!newStory.text && !newStory.bucketName) {
+        res.status(400).send({ error:true, message: 'Please provide text or bucket name or both.' });
+    } else {
+        storyModel.createStory(newStory, function(err, record) {
```
Comment (yorhodes) [inline server/controllers/storyController.js]: looks great
maybe in future we can have separate functions for 
1. validating request body properties
2. performing application logic
3. raising appropriate status code and sending response

File: server/docker-compose.yml [hunk 4]
```diff
@@ -1,22 +0,0 @@
-version: "2"
```
Comment (yorhodes) [inline server/docker-compose.yml]: any reason to delete this?
want to make sure team members' toes aren't being stepped on

File: server/server.js [hunk 5]
```diff
@@ -1,29 +1,36 @@
...
+app.post('/story', story.createStory);
+app.get('/story', story.getStory);
+
+// example payload: {coord: {lat: 5, lng: 50}}
+app.post('/test-middleware', validationMiddleware.isValidCoordinate, db.getTest);
 
 app.listen(port, () => {
-  console.log(`App running on port ${port}.`)
-})
+    console.log(`App running on port ${port}.`);
```
Comment (yorhodes) [inline server/server.js]: maybe we should make a style guide or add a linter for consistency

File: server/controllers/storyController.js (hunk 3 above)
Comment (XinyangZhu) [reply server/controllers/storyController.js]: good point! i will look into that and separate those validation functions in the future. this one was solely for testing that the middleware was functioning as expected.

**Pull request merged**